internal/storage/field_value.go

```diff
@@ -1029,17 +1031,43 @@ func (ifv *FloatVectorFieldValue) Size() int64 {
 	return int64(len(ifv.Value) * 8)
 }
 
-func NewScalarFieldValueFromGenericValue(dtype schemapb.DataType, gVal *planpb.GenericValue) ScalarFieldValue {
+func NewScalarFieldValueFromGenericValue(dtype schemapb.DataType, gVal *planpb.GenericValue) (ScalarFieldValue, error) {
 	switch dtype {
+	case schemapb.DataType_Int8:
+		i64Val := gVal.Val.(*planpb.GenericValue_Int64Val)
+		if i64Val.Int64Val > math.MaxInt8 || i64Val.Int64Val < math.MinInt8 {
```

Comment:
is this check necessary? what if I set a INT8 clustering key and query with a expr out of INT8 range. for example: k< INT8.maxValue + 1. This should be a valid query?